apps/middleman/src/lib/dal/nodes.ts

```diff
@@ -1,9 +1,11 @@
 import "server-only";
 import { db } from "@/db";
-import { desc } from 'drizzle-orm'
+import {desc, eq} from 'drizzle-orm'
+import {nodesTable} from "@/db/schema";
 
-export async function getNodesByUser() {
-  return await db.query.nodesTable.findMany({
+export async function getNodesByUser(userIdentity: string) {
+  return db.query.nodesTable.findMany({
+    where: eq(nodesTable.createdBy, userIdentity),
```

Comment:
Is there a reason why you didn't use a function like `getCurrentUserIdentity` (the one you use in the provider to get the user identity) instead of passing the user identity as a param?

In the end the server actions are endpoints, so we must control the access to the users data.

Reply:
Same as above.